pkg/maps/lbmap/lbmap.go

```diff
@@ -68,6 +73,30 @@ func (*LBBPFMap) UpsertService(
 
 	slot := 1
 	svcVal := svcKey.NewValue().(ServiceValue)
+
+	if useMaglev && len(backends) != 0 {
+		var backendIDs []uint16
+
+		backendNames := make([]string, 0, len(backends))
+		for name := range backends {
+			backendNames = append(backendNames, name)
+		}
+		m := uint64(option.Config.MaglevTableSize)
+		table := maglev.GetLookupTable(backendNames, m)
+		backendIDs = make([]uint16, m)
```

Comment:
Is there a memory allocations consideration here? I get the impression that `m` could end up somewhat large so if we reshuffled backends often, it might start to influence memory usage. I guess Tobias would be the one with the most recent understanding of potential impact here.

Reply:
Good point. I might be mistaken, but it looks like `(*LBBPFMap).UpsertServer` isn't called at regular intervals or in a loop, but only when services are added/updated. Given this and the fact that `backendIDs` is rather short-lived here, I wouldn't expect a big impact even if `m` is rather large. But it might still be worth checking memory usage with Maglev enabled to confirm.

Reply:
I'm wondering how expensive is allocation? Maybe it makes sense to have a preallocated `([]uint16,m)`? The default `m` is , 16381, and there is only one goroutine calling `UpsertService(...)`.

Reply:
> Good point. I might be mistaken, but it looks like (*LBBPFMap).UpsertServer isn't called at regular intervals or in a loop, but only when services are added/updated.

If we're dealing with backends, then presumably also service scale up / scale down, app redeployment, etc. then it may not necessarily be constant, but it could be somewhat frequent.

I vaguely recall discussion before about Go allocating memory over time, growing and growing and not releasing it unless there is pressure from the system. I clearly didn't fully internalize that discussion so please do point out any poor assumptions here, but let me ask: if you have 1K pods that are backends to services which you shuffle once per day (for updates or regular scaling or whatever else), and let's say you have a few KB per invocation of this code path, then over time will this cause Go memory usage to climb if it doesn't have pressure to release the memory again? Or would it more likely end up reusing the same memory region each time this path is invoked?

It may turn out that the answer is that this is insignificant. I don't have any specific knowledge to point that we must be careful here. I mainly raised it for the discussion and figure out whether we think it's likely we think it's worth investigating further here. Either way, the goal of this PR overall is adding a new feature. Presumably if the feature isn't enabled it shouldn't affect current memory usage, so I don't think this is something that should block the overall feature.

Reply:
@joestringer I think you are referring to the Go's GC being conservative about releasing reclaimed memory back to OS. Unfortunately, I cannot find an issue / article documenting this behavior. Maybe @tklauser knows it.

Anyway, as the `m` doesn't change during the runtime (and currently is the same for all services) AND `UpsertService()` invocation is serialized, it makes sense to preallocate the buffer. Added to the follow up issue.